internal/expr/bindtypes.go

```diff
@@ -378,6 +416,50 @@ func (e *outputExpr) bindTypes(argInfo typeinfo.ArgInfo) (te any, err error) {
 	return toe, nil
 }
 
+// valueAccessor defines an accessor that can be used to generate a typedColumn
+// with the given column name.
+type valueAccessor interface {
+	typedColumn(argInfo typeinfo.ArgInfo, columnName string) (typedColumn, error)
```

Comment:
If I am reading this correctly, `typedColumn` is only used for insert columns? (context: memberAccessor below returns a `newInsertColumn`.

Reply:
And literals

Reply:
literals in insert expressions, right?

Reply:
Yea, right now the valueAccessor is implemented by `memberAccessor` and `literal` the typedColumn interface is implemented by `literalColumn` and `insertColumn`.

Its not the best abstraction, but I was struggling to find a nice way to do literals.